spring-boot-admin-server/src/test/java/de/codecentric/boot/admin/notify/microsoft/teams/MicrosoftTeamsNotifierTest.java

```diff
@@ -0,0 +1,231 @@
+
+        Object expected = testMTNotifier.getRegisteredMessage(stubApp);
+
+        testMTNotifier.doNotify(registeredEvent);
+
+        verify(mockRestTemplate).postForObject(any(URI.class), eq(expected), eq(Object.class));
+    }
+
+    @Test
+    public void test_onApplicationStatusChangedEventFromDownToUp_resolve() throws Exception {
+        ClientApplicationStatusChangedEvent statusChangedEvent = new ClientApplicationStatusChangedEvent(stubApp,
+                StatusInfo.ofDown(), StatusInfo.ofUp());
+
+        Object expected = testMTNotifier.getStatusChangedMessage(stubApp, StatusInfo.ofDown(), StatusInfo.ofUp());
+
+        testMTNotifier.doNotify(statusChangedEvent);
+
+        verify(mockRestTemplate).postForObject(any(URI.class), eq(expected), eq(Object.class));
+    }
+
+    @Test
+    public void test_onApplicationStatusChangedEventFromUpToDown_resolve() throws Exception {
+        ClientApplicationStatusChangedEvent statusChangedEvent = new ClientApplicationStatusChangedEvent(stubApp,
+                StatusInfo.ofUp(), StatusInfo.ofDown());
+
+        Object expected = testMTNotifier.getStatusChangedMessage(stubApp, StatusInfo.ofUp(), StatusInfo.ofDown());
+
+        testMTNotifier.doNotify(statusChangedEvent);
+
+        verify(mockRestTemplate).postForObject(any(URI.class), eq(expected), eq(Object.class));
+    }
+
+    @Test
+    public void test_onApplicationStatusChangedEventFromUpToOffline_resolve() throws Exception {
+        ClientApplicationStatusChangedEvent statusChangedEvent = new ClientApplicationStatusChangedEvent(stubApp,
+                StatusInfo.ofUp(), StatusInfo.ofOffline());
+
+        Object expected = testMTNotifier.getStatusChangedMessage(stubApp, StatusInfo.ofUp(), StatusInfo.ofOffline());
+
+        testMTNotifier.doNotify(statusChangedEvent);
+
+        verify(mockRestTemplate).postForObject(any(URI.class), eq(expected), eq(Object.class));
+    }
+
+    @Test
+    public void test_onApplicationStatusChangedEventFromDownToOffline_resolve() throws Exception {
+        ClientApplicationStatusChangedEvent statusChangedEvent = new ClientApplicationStatusChangedEvent(stubApp,
+                StatusInfo.ofDown(), StatusInfo.ofOffline());
+
+        Object expected = testMTNotifier.getStatusChangedMessage(stubApp, StatusInfo.ofDown(), StatusInfo.ofOffline());
+
+        testMTNotifier.doNotify(statusChangedEvent);
+
+        verify(mockRestTemplate).postForObject(any(URI.class), eq(expected), eq(Object.class));
+    }
+
+    @Test
+    public void test_getDeregisteredMessageForAppReturns_correctContent() {
+        Message testMessage = testMTNotifier.getDeregisteredMessage(stubApp);
+
+        assertEquals("Title doesn't match", DE_REGISTERED_TITLE, testMessage.getTitle());
+        assertEquals("Summary doesn't match", MESSAGE_SUMMARY, testMessage.getSummary());
+
+        Section expectedSection = new Section();
```

Comment:
Instead of creating a expected section and comparing the actual against it, please assert the attributes of the actual section against their expected values. This way you see direct which attribute differs instead of just knowing that some of the many attributes is wrong! (and also you don't need to implement and test equals/hashcode)